src/components/ItemList/index.js

```diff
@@ -80,6 +80,38 @@ export function ItemList({ items }) {
         alert(err)
       })
   }
+
+  function AuthedBuyClick(itemIndex, authedflag) {
```

Comment:
`authedflag` here seems unnecessary since we can directly use `auth` at line 204.
This will require to move the rendering functions lower to the line.